biom/table.py

```diff
@@ -3543,7 +3554,7 @@ def axis_dump(grp, ids, md, group_md, order, compression=None):
                 # is cleaner, as is the parse
                 grp.create_dataset('ids', shape=(len_ids,),
                                    dtype=H5PY_VLEN_STR,
-                                   data=[str(i) for i in ids],
+                                   data=[str(i).encode('utf8') for i in ids],
```

Comment:
This will do strange things (from the user's perspective) if the user hands ids as `byte`s in Python 3, but they really shouldn't do that and it works entirely as expected otherwise.